spec/requests/rack_attack_spec.rb

```diff
@@ -0,0 +1,169 @@
+require 'rails_helper'
+
+describe 'throttling requests' do
+  include Rack::Test::Methods
+
+  def app
+    Rails.application
+  end
+
+  describe 'whitelists' do
+    it 'allows all requests from localhost' do
+      get '/'
+
+      expect(last_request.env['rack.attack.throttle_data']).to be_nil
+    end
+  end
+
+  describe 'high requests per ip' do
+    it 'reads the limit and period from ENV vars' do
+      get '/', {}, 'REMOTE_ADDR' => '1.2.3.4'
+
+      data = {
+        count: 1,
+        limit: Figaro.env.requests_per_ip_limit.to_i,
+        period: Figaro.env.requests_per_ip_period.to_i.seconds
+      }
+
+      expect(last_request.env['rack.attack.throttle_data']['req/ip']).to eq(data)
+    end
+
+    context 'when the number of requests is lower than the limit' do
+      it 'does not throttle' do
+        2.times do
+          get '/', {}, 'HTTP_X_FORWARDED_FOR' => '1.2.3.4'
+        end
+
+        expect(last_response.status).to_not eq(429)
+      end
+    end
+
+    context 'when the request is for an asset' do
+      it 'does not throttle' do
+        4.times do
+          get '/assets/application.js', {}, 'HTTP_X_FORWARDED_FOR' => '1.2.3.4'
+        end
+
+        expect(last_response.status).to_not eq(429)
+      end
+    end
+
+    context 'when the number of requests is higher than the limit' do
+      before do
+        allow(Rails.logger).to receive(:warn)
+
+        4.times do
```

Comment:
Hmm. I disagree. By looking at the `rack_attack.rb` file, one can see that the settings are configured via Figaro, so it should be clear that the settings are configured in `application.yml`. I'd rather be able to set values for testing in one place rather than having to run 4 different `allow(Figaro.env).to receive(:foo).and_return('bar')` statements for each spec.